identity_storage/src/key_id_storage/memstore.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::key_id_storage::key_id_storage_error::KeyIdStorageError;
+use crate::key_id_storage::key_id_storage_error::KeyIdStorageErrorKind;
+use crate::key_id_storage::storage::KeyIdStorage;
+use crate::key_storage::shared::Shared;
+use crate::key_storage::KeyId;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use tokio::sync::RwLockReadGuard;
+use tokio::sync::RwLockWriteGuard;
+
+use super::method_digest::MethodDigest;
+use super::storage::KeyIdStorageResult;
+
+type KeyIdStore = HashMap<MethodDigest, KeyId>;
+
+/// An insecure, in-memory [`KeyIdStorage`] implementation that serves as an example and may be used in tests.
+#[derive(Debug)]
+pub struct KeyIdMemstore {
+  key_id_store: Shared<KeyIdStore>,
+}
+
+impl KeyIdMemstore {
+  /// Creates a new, empty `KeyIdMemstore` instance.
+  pub fn new() -> Self {
+    Self {
+      key_id_store: Shared::new(HashMap::new()),
+    }
+  }
+}
+
+impl Default for KeyIdMemstore {
+  fn default() -> Self {
+    Self::new()
+  }
+}
+
+#[cfg_attr(not(feature = "send-sync-storage"), async_trait(? Send))]
+#[cfg_attr(feature = "send-sync-storage", async_trait)]
+impl KeyIdStorage for KeyIdMemstore {
+  async fn insert_key_id(&self, key: MethodDigest, value: KeyId) -> KeyIdStorageResult<()> {
+    let mut key_id_store: RwLockWriteGuard<'_, KeyIdStore> = self.key_id_store.write().await;
+    key_id_store.insert(key, value);
```

Comment:
This doesn't return an error if the key already exists, which is different from the requirements described on `KeyIdStorage::insert_key_id`. Adding a test case for this would help catch this error.